code/modules/ruins/objects_and_mobs/necropolis_gate.dm

```diff
@@ -60,3 +60,168 @@
 		. = ..()
 	else
 		return QDEL_HINT_LETMELIVE
+
+//stone tiles for boss arenas
+/obj/structure/stone_tile
+	name = "stone tile"
+	icon = 'icons/turf/boss_floors.dmi'
+	icon_state = "pristine_tile1"
+	layer = OVER_LATTICE_LAYER
+	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | UNACIDABLE | ACID_PROOF
+	var/tile_key = "pristine_tile"
+	var/tile_random_sprite_max = 24
+
+/obj/structure/stone_tile/Initialize(mapload)
+	. = ..()
+	icon_state = "[tile_key][rand(1, tile_random_sprite_max)]"
+
+/obj/structure/stone_tile/block
+	name = "stone block"
+	icon_state = "pristine_block1"
+	tile_key = "pristine_block"
+	tile_random_sprite_max = 4
+
+/obj/structure/stone_tile/slab
+	name = "stone slab"
+	icon_state = "pristine_slab1"
+	tile_key = "pristine_slab"
+	tile_random_sprite_max = 4
+
+/obj/structure/stone_tile/center
+	name = "stone center tile"
+	icon_state = "pristine_center1"
+	tile_key = "pristine_center"
+	tile_random_sprite_max = 4
+
+/obj/structure/stone_tile/surrounding
+	name = "stone surrounding slab"
+	icon_state = "pristine_surrounding1"
+	tile_key = "pristine_surrounding"
+	tile_random_sprite_max = 2
+
+/obj/structure/stone_tile/surrounding_tile
+	name = "stone surrounding tile"
+	icon_state = "pristine_surrounding_tile1"
+	tile_key = "pristine_surrounding_tile"
+	tile_random_sprite_max = 2
+
+//cracked stone tiles
+/obj/structure/stone_tile/cracked
```

Comment:
crossing a cracked tile should make it animate into the floor (pixel_z iirc) and then delete itself.
you know, classic video game stuff.